src/Cardano/AddressDerivation.hs

```diff
@@ -202,3 +223,48 @@ class GenMasterKey (key :: Depth -> * -> *) where
 
     -- | Generate a root key from a corresponding seed.
     genMasterKey :: GenMasterKeyFrom key -> ScrubbedBytes -> key 'RootK XPrv
+
+
+-- | Address is mere wrapper around a ByteString and represents an encoded address.
+newtype Address = Address
+    { unAddress :: ByteString
+    } deriving (Show, Generic, Eq, Ord)
+
+instance NFData Address
+
+-- | Available network options.
+data NetworkDiscriminant = Mainnet | Testnet Nat
+
+class NetworkDiscriminantVal (n :: NetworkDiscriminant) where
+    networkDiscriminantVal :: Text
+
+instance NetworkDiscriminantVal 'Mainnet where
+    networkDiscriminantVal =
+        "mainnet"
+
+instance KnownNat pm => NetworkDiscriminantVal ('Testnet pm) where
+    networkDiscriminantVal =
+        "testnet (" <> T.pack (show $ natVal $ Proxy @pm) <> ")"
+
+-- | Magic constant associated to a given network
+newtype ProtocolMagic = ProtocolMagic { getProtocolMagic :: Int32 }
```

Comment:
Note sure why we are using an `Int32` here rather than a `Word32` :thinking:  ?

Reply:
yes, changed to Word32 and cbor encoders/decoders also respectively